tests/Tokenizer/Transformer/DisjunctiveNormalFormTypeParenthesisTransformerTest.php

```diff
@@ -0,0 +1,290 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Tokenizer\Transformer;
+
+use PhpCsFixer\Tests\Test\AbstractTransformerTestCase;
+use PhpCsFixer\Tokenizer\CT;
+
+/**
+ * @covers \PhpCsFixer\Tokenizer\Transformer\DisjunctiveNormalFormTypeParenthesisTransformer
+ *
+ * @internal
+ *
+ * @requires PHP 8.2
+ */
+final class DisjunctiveNormalFormTypeParenthesisTransformerTest extends AbstractTransformerTestCase
+{
+    /**
+     * @param array<int, int> $expectedTokens
+     *
+     * @dataProvider provideProcessCases
+     */
+    public function testProcess(string $source, array $expectedTokens): void
+    {
+        $this->doTest($source, $expectedTokens, [CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN, CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE]);
+    }
+
+    public static function provideProcessCases(): iterable
```

Comment:
Maybe we should include invokable class in test case? There are a lot of `(`+`)` pairs that shouldn't be matched as DNFs - [example](https://3v4l.org/aTb7e).

```php
interface Foo {}
interface Bar {}

class A {
    public function __invoke((Foo&Bar)|string $a): string {
        return 'A';
    }
}

return (new A())(new class implements Foo, Bar {});
```